### PR TITLE
[DEV-6922] Fix outlay calculations

### PR DESCRIPTION
**Pull request by emily-brents**
**Description:**
Changes the method used to calculate how the outlay amounts are calculated on the COVID-19 profile page for endpoints using the Accounts ES index

**Technical details:**
Instead of just summing the `gross_outlay_amount_by_award_cpe` field, we also add the `ussgl487200_down_adj_pri_ppaid_undel_orders_oblig_refund_cpe` and `ussgl497200_down_adj_pri_paid_deliv_orders_oblig_refund_cpe` fields
**Requirements for PR merge:**

1. [X] Unit & integration tests updated
2. [X] API documentation updated
3. [ ] Necessary PR reviewers:
    - [ ] Backend
4. [X] Matview impact assessment completed
5. [X] Frontend impact assessment completed
6. [X] Data validation completed
7. [ ] Appropriate Operations ticket(s) created
8. [X] Jira Ticket [DEV-6922](https://federal-spending-transparency.atlassian.net/browse/DEV-6922):
    - [x] Link to this Pull-Request
    - [ ] Performance evaluation of affected (API | Script | Download)
    - [ ] Before / After data comparison

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

File: usaspending_api/disaster/tests/fixtures/federal_account_data.py [hunk 1]
```diff
@@ -162,6 +162,8 @@ def generic_account_data():
         disaster_emergency_fund=defc_m,
         treasury_account=tre_acct1,
         distinct_award_key="0wefjwe|3443r||",
+        ussgl487200_down_adj_pri_ppaid_undel_orders_oblig_refund_cpe=0,
```
Comment (tony-sappe) [inline usaspending_api/disaster/tests/fixtures/federal_account_data.py:165]: Without non-zero values in mock data, are the tests really testing the new logic?

File: usaspending_api/etl/elasticsearch_loader_helpers/transform_data.py [hunk 2]
```diff
@@ -97,7 +97,11 @@ def transform_covid19_faba_data(worker: TaskSpec, records: List[dict]) -> List[d
         generated_unique_award_id = record.pop("generated_unique_award_id")
         total_loan_value = record.pop("total_loan_value")
         obligated_sum = record.get("transaction_obligated_amount") or 0  # record value for key may be None
-        outlay_sum = record.get("gross_outlay_amount_by_award_cpe") or 0  # record value for key may be None
+        outlay_sum = (
+            (record.get("gross_outlay_amount_by_award_cpe") or 0)
+            + (record.get("ussgl487200_down_adj_pri_ppaid_undel_orders_oblig_refund_cpe") or 0)
+            + (record.get("ussgl497200_down_adj_pri_paid_deliv_orders_oblig_refund_cpe") or 0)
+        )  # record value for key may be None
```
Comment (tony-sappe) [inline usaspending_api/etl/elasticsearch_loader_helpers/transform_data.py:104]:
```suggestion
        )  # record value for any key may be None
```

File: usaspending_api/disaster/v2/views/elasticsearch_account_base.py [hunk 3]
```diff
@@ -110,8 +110,10 @@ def build_elasticsearch_search_with_aggregations(self) -> Optional[AccountSearch
         )
         sum_covid_outlay = A(
             "sum",
-            field="financial_accounts_by_award.gross_outlay_amount_by_award_cpe",
-            script={"source": "doc['financial_accounts_by_award.is_final_balances_for_fy'].value ? _value : 0"},
+            script="""doc['financial_accounts_by_award.is_final_balances_for_fy'].value ? (
+             ( doc['financial_accounts_by_award.gross_outlay_amount_by_award_cpe'].size() > 0 ? doc['financial_accounts_by_award.gross_outlay_amount_by_award_cpe'].value : 0)
```
Comment (tony-sappe) [inline usaspending_api/disaster/v2/views/elasticsearch_account_base.py:114]: Checking the .size() for each field looks rough. Is this the recommended way to handle nulls in aggregate fields? Couldn't `outlay_sum` be used because the ETL is storing all of the values in the single field?

**Review by tony-sappe — CHANGES_REQUESTED** [review]
Would be good to include some non-zero mock data for the new fields in the business logic

File: usaspending_api/disaster/v2/views/elasticsearch_account_base.py (hunk 3 above)
Comment (emily-brents) [reply usaspending_api/disaster/v2/views/elasticsearch_account_base.py:114]: checking .size() is the way the elasticsearch docs recommend to check for existing data. We can't use outlay sum due to an edge case - if we have an award with multiple DEFC, and we filter by just "L", for example, we can't use the outlay_sum without including the data from other DEFC.

File: usaspending_api/disaster/tests/fixtures/federal_account_data.py (hunk 1 above)
Comment (emily-brents) [reply usaspending_api/disaster/tests/fixtures/federal_account_data.py:165]: I created new mock data for the outlay tests just to make the changes more obvious for those tests, I can go back and add data and alter the existing tests if you prefer.

File: usaspending_api/etl/elasticsearch_loader_helpers/transform_data.py (hunk 2 above)
Comment (emily-brents) [reply usaspending_api/etl/elasticsearch_loader_helpers/transform_data.py:104]: yeah that's fair

File: usaspending_api/disaster/v2/views/elasticsearch_account_base.py (hunk 3 above)
Comment (tony-sappe) [reply usaspending_api/disaster/v2/views/elasticsearch_account_base.py:114]: Is `outlay_sum` used anywhere?

File: usaspending_api/disaster/tests/fixtures/federal_account_data.py (hunk 1 above)
Comment (tony-sappe) [reply usaspending_api/disaster/tests/fixtures/federal_account_data.py:165]: Ah, I overlooked the intentions of the new tests. When I was working on the hotfix changes it was pretty easy to alter the mock data values to still equal the original result values. Fine if you want to leave the tests as-is

File: usaspending_api/disaster/v2/views/elasticsearch_account_base.py (hunk 3 above)
Comment (emily-brents) [reply usaspending_api/disaster/v2/views/elasticsearch_account_base.py:114]: Looks like it is not

**Review by tony-sappe — APPROVED** [review]
Approving the PR, but realize changes might be made. Up to @emily-brents
